r1cs/gadgets/std/src/bits/uint64.rs

```diff
@@ -393,22 +393,19 @@ impl<ConstraintF: Field> CondSelectGadget<ConstraintF> for UInt64 {
             .zip(&false_value.bits)
             .enumerate()
             .map(|(i, (t, f))| {
-                Boolean::conditionally_select(&mut cs.ns(|| format!("bit {}", i)), cond, t, f)
-            });
-        let mut bits = [Boolean::Constant(false); 64];
-        for (result, new) in bits.iter_mut().zip(selected_bits) {
-            *result = new?;
-        }
-
-        let value = cond.get_value().and_then(|cond| {
-            if cond {
+                Boolean::conditionally_select(&mut cs.ns(|| format!("select bit {}", i)), cond, t, f)
+            })
+            .collect::<Result<_, _>>()?;
+        
+        let value = 
+            if cond.get_value().get()? {
```

Comment:
This version of conditionally_select creates an issue on the setup phase of the circuit. I don't have the issue using the previous version "let value = cond.get_value().and_then(|cond| { ..."

Reply:
It might be a leftover of the commits revert. You're right. Can you restore it ?

Reply:
Done